feat: sequence task workforce

nitpicker55555 opened the pull request:
## Description

Support for processing a sequence of tasks in the `Workforce`, where each task can share context (e.g., results, memory) with others. 

- Added `process_task_sequence` and `process_task_sequence_async` methods to `Workforce`
- Introduced `shared_context` and `shared_channel` to persist information across tasks
- Each task's result is stored in `shared_context` with the pattern `task_{task.id}`
- Compatible with current single-task flow; does not affect `process_task`

This addresses collaborative task execution as proposed in [Issue #1832](https://github.com/camel-ai/camel/issues/1832).

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have linked this PR to an issue using the Development section or `Fixes #1832` (**required**)
- [x] I have checked if any dependencies need to be added or updated in `pyproject.toml` and `uv lock`
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation if needed
- [x] I have added examples if this is a new feature

zjrwtx commented on examples/workforce/task_sequence_hackathon_judges.py:
```diff
@@ -0,0 +1,153 @@
+                "technology used, innovation, and potential impact."
+            ),
+        ),
+        Task(
```
can we add some comments on these?

zjrwtx replied:
like step by step run all task ....

zjrwtx commented:
thanks @nitpicker55555 ,left some comment

Wendong-Fan reviewed (changes requested):
Thanks @nitpicker55555  for the contribution, and apologies for the delayed review comment.

The current implementation appears to process a sequence of top-level tasks within the workforce, rather than allowing for a pre-defined, fixed sequence of subtasks for a single overarching goal. As it stands, each task in the sequence is still subject to the workforce's internal decomposition process.

The user's requirement, as I understand it, is to support a scenario where subtasks are explicitly defined and executed in a specific order without further decomposition by the workforce. This would give more direct control over the execution flow.

Maybe allowing the Task to specify which worker node should handle it could offer more granular control. Happy to discuss this further

nitpicker55555 commented:
@Wendong-Fan I've updated the Workforce so it now processes pre-defined subtasks directly instead of always decomposing them, and I also added a way for user to assign specific subtasks to the exact worker user wants.  And I've put multi_hop_example.py to show how it all works, looking forward to your feedback

Wendong-Fan commented:
<img width="354" alt="Screenshot 2025-06-06 at 17 30 16" src="https://github.com/user-attachments/assets/92f22cd9-6c5a-4880-8756-18682ab35045" />


> @Wendong-Fan I've updated the Workforce so it now processes pre-defined subtasks directly instead of always decomposing them, and I also added a way for user to assign specific subtasks to the exact worker user wants. And I've put multi_hop_example.py to show how it all works, looking forward to your feedback

thanks @nitpicker55555 ! Will check the update later, please remember to update the PR label

Wendong-Fan reviewed (commented):
tested with the example code seems it's hanging, could @nitpicker55555 help checking the reason?

nitpicker55555 commented:
@Wendong-Fan CI passed now, this failure happened many times because of the test_alpaca_collector file, 
this is a test that involves using an LLM to construct data in the Alpaca format. One of the assertions checks that the input key (which represents additional background information) is empty. However, the LLM may incorrectly interpret the original user question as the input, leading to assertion failures. 
Another contributing factor is that the source code in alpaca_collector does not include examples where input is non-empty, which may increase the likelihood of misclassification by the LLM.

Wendong-Fan commented:
> @Wendong-Fan CI passed now, this failure is not related to current pr and happened many times because of the test_alpaca_collector file, this is a test that involves using an LLM to construct data in the Alpaca format. One of the assertions checks that the input key (which represents additional background information) is empty. However, the LLM may incorrectly interpret the original user question as the input, leading to assertion failures. Another contributing factor is that the source code in alpaca_collector does not include examples where input is non-empty, which may increase the likelihood of misclassification by the LLM.

hey @nitpicker55555 , I mean it was hanging when I ran `examples/workforce/multi_hop_example.py`
